app/components/templates/form.tsx

```diff
@@ -0,0 +1,255 @@
+          autoFocus
+          onChange={updateTitle}
+          className="w-full"
+          defaultValue={name || ""}
+          placeholder="Booking Arrangement 2023"
+          required={zodFieldIsRequired(NewTemplateFormSchema.shape.name)}
+        />
+      </FormRow>
+      <FormRow
+        rowLabel="Type"
+        className="border-b-0 pb-0"
+        required={zodFieldIsRequired(NewTemplateFormSchema.shape.type)}
+      >
+        <Select
+          name={zo.fields.type()}
+          defaultValue={selectedType}
+          disabled={disabled}
+          onValueChange={(val) =>
+            setSelectedType(
+              val.toString() === TemplateType.BOOKINGS
+                ? TemplateType.BOOKINGS
+                : TemplateType.CUSTODY
+            )
+          }
+        >
+          <SelectTrigger
+            disabled={isEdit}
+            className="px-3.5 py-3"
+            placeholder="Choose a field type"
+          >
+            <SelectValue />
+          </SelectTrigger>
+          <SelectContent
+            position="popper"
+            className="w-full min-w-[300px]"
+            align="start"
+          >
+            <div className=" max-h-[320px] overflow-auto">
+              {[TemplateType.BOOKINGS, TemplateType.CUSTODY].map((value) => (
+                <SelectItem value={value} key={value}>
+                  <span className="mr-4 text-[14px] text-gray-700">
+                    {value}
+                  </span>
+                </SelectItem>
+              ))}
+            </div>
+          </SelectContent>
+        </Select>
+      </FormRow>
+      <FormRow
+        required={false}
+        rowLabel="Ask for signature"
+        subHeading={<p>Users will be asked to sign document</p>}
+        className="border-b-0 pb-0"
+      >
+        <Switch
+          required={false}
+          disabled={disabled}
+          defaultChecked={signatureRequired}
+        />
+      </FormRow>
+      <FormRow
+        className="border-0 pb-0"
+        rowLabel="Description"
+        required={false}
+      >
+        <Input
+          label="Description"
+          hideLabel
+          inputType="textarea"
+          name={zo.fields.description()}
+          disabled={disabled}
+          error={zo.errors.description()?.message}
+          className="w-full border-b-0 pb-0"
+          defaultValue={description || ""}
+          placeholder="Store the booking arrangement for 2023"
+          required={false}
+        />
+      </FormRow>
+      <FormRow required={!isEdit} rowLabel="Upload PDF">
+        <div>
+          <p className="hidden lg:block">Accepts PDF (max. 5 MB)</p>
+          <Input
+            required={!isEdit}
+            disabled={disabled}
+            accept="application/pdf,.pdf"
+            name={"pdf"}
+            type="file"
+            size={5_000_000}
+            onChange={handleFileChange}
+            label={""}
+            hideLabel
+            error={fileError}
```

Comment:
1. the validation is taking place in the `action` function of the creat new template component. I am however sending an error. But I have no idea about how I should display it. I feel this can be a lot easier if we do the validation in the form itself, but that too isn't happening.

2. Was unsure about this, I'll fix that

Reply:
P.S. went through [this](https://github.com/colinhacks/zod/issues/387) article. seems like there is a problem with zod file input. A little bit of help would be really appreciated

Reply:
Okey so before we never validated files using zod. I am not sure why, maybe due to laziness maybe some other reason, but in this case that we have required files, I think its nice to create a function that can help us validate zod file fields. You can then manage the error on the front-end like with any other zod field.
So please take a look at this: https://github.com/colinhacks/zod/issues/387#issuecomment-1191390673

It provides a full schema that validates type, size, and that its present.